dokka-subprojects/analysis-kotlin-api/src/test/kotlin/org/jetbrains/dokka/analysis/test/jvm/java/InnerModifierJavaAnalysisTest.kt

```diff
@@ -0,0 +1,293 @@
+ */
+
+package org.jetbrains.dokka.analysis.test.jvm.java
+
+import org.jetbrains.dokka.analysis.test.api.javaTestProject
+import org.jetbrains.dokka.analysis.test.api.parse
+import org.jetbrains.dokka.model.*
+import org.jetbrains.dokka.model.properties.WithExtraProperties
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+class InnerModifierJavaAnalysisTest {
+
+    @Test
+    fun `top level java declarations should not have kotlin inner modifier`() {
+        val testProject = javaTestProject {
+            javaFile(pathFromSrc = "JavaClass.java") {
+                +"public class JavaClass { }"
+            }
+            javaFile(pathFromSrc = "JavaEnum.java") {
+                +"public enum JavaEnum { S; }"
+            }
+            javaFile(pathFromSrc = "JavaInterface.java") {
+                +"public interface JavaInterface { }"
+            }
+            javaFile(pathFromSrc = "JavaAnnotation.java") {
+                +"public @interface JavaAnnotation { }"
+            }
+        }
+
+        val module = testProject.parse()
+        val pkg = module.packages.single()
+
+        assertEquals(4, pkg.classlikes.size)
+
+        pkg.classlikes.forEach {
+            @Suppress("UNCHECKED_CAST")
+            assertTrue((it as WithExtraProperties<Documentable>).kotlinOnlyModifiers().isEmpty())
+        }
+    }
+
+    @Test
+    fun `java declarations nested inside interfaces should not have kotlin inner modifier`() {
+        val testProject = javaTestProject {
+            javaFile(pathFromSrc = "JavaInterface.java") {
+                +"""
+                    public interface JavaInterface {
+                        public class InnerClass { }
+                        public static class NestedClass { }
+                        public enum InnerEnum {S}
+                        public static enum NestedEnum {S}
+                        public interface InnerInterface { }
+                        public static interface NestedInterface { }
+                        public @interface InnerAnnotation { }
+                        public static @interface NestedAnnotation { }
+                    }
+                """
+            }
+        }
+
+        val module = testProject.parse()
+        val pkg = module.packages.single()
+        val javaInterface = pkg.classlikes.single()
+
+        assertTrue(javaInterface is DInterface)
+        assertEquals(8, javaInterface.classlikes.size)
+
+        javaInterface.classlikes.forEach {
+            @Suppress("UNCHECKED_CAST")
+            assertTrue((it as WithExtraProperties<Documentable>).kotlinOnlyModifiers().isEmpty())
+        }
+    }
+
+    @Test
+    fun `java declarations nested inside annotation interface should not have kotlin inner modifier`() {
+        val testProject = javaTestProject {
+            javaFile(pathFromSrc = "JavaAnnotation.java") {
```

Comment:
btw, for some reason I didn't expect Java to handle root packages correctly, but apparently it works :) 

![2023-11-22_02-42-14](https://github.com/Kotlin/dokka/assets/22685910/3b8b3e34-0f92-4f26-bb8c-3222e88c1c10)